weave/weave_client.py

```diff
@@ -662,8 +667,13 @@ def _save_object_basic(
         return ref
 
     def _save_nested_objects(self, obj: Any, name: Optional[str] = None) -> Any:
-        if get_ref(obj) is not None:
-            return
+        ref = get_ref(obj)
+        if ref is not None:
+            if ref.project == self.project:
+                return
```

Comment:
Why do you return here?  Is it because the obj is already saved in the other project?

Reply:
Its essentially a cached ref, no need to make a new ref if the object already has one. This is currently the case for anytime the obj has a ref, we are just adding the check for when the projects don't match @andrewtruong

Reply:
I would add that as a comment

Reply:
yep